### PR TITLE
feat: add browser notification

### PR DESCRIPTION
**Pull request by dni**
used when a swap is claimed or refunded.

screenshot-1711525563

screenshot-1714982106

now it  is opt in only via settings

### CODE REVIEW

**Review by kilrau — CHANGES_REQUESTED** [review]
I open the landing page and it immediately asks for notification permissions - NO
image

Then again on "Create Swap"... 

Please read the issue again! We only want notification permissions *if user toggles 0-conf off*: https://github.com/BoltzExchange/boltz-web-app/issues/346 so that should be the trigger.

**Comment by kilrau** [comment]
As our testing just revealed: browser notification is "in your face" on page load, only should be triggered on toggle

**Comment by kilrau** [comment]
In your face permission request gone, please resolve conflicts and request review, I'll do some e2e testing @dni

**Review by kilrau — CHANGES_REQUESTED** [review]
- need to gracefully handle incognito window
[Screencast from 2024-05-22 13-26-16.webm](https://github.com/BoltzExchange/boltz-web-app/assets/19181985/2ca3be57-952a-4a52-b67a-6841d7055a75)
- didn't get any notification when I was in a different tab while the swap completed even though notifications were turned on

Tested in FF & Chromium.

**Comment by kilrau** [comment]
Edge case but if the fix is easy let's please do it: if I withdraw notification permissions while switch is on, it's still on after reload. Reload should check permissions I guess.

[Screencast from 2024-05-22 15-43-46.webm](https://github.com/BoltzExchange/boltz-web-app/assets/19181985/5b65001e-0af9-4f20-b01f-a4e83595a719)

**Review by kilrau — CHANGES_REQUESTED** [review]
We fixed a couple of things in my regtest setup and notfications are working now.

TODOs:

1. what I mentioned [above](https://github.com/BoltzExchange/boltz-web-app/pull/528#issuecomment-2124840924):
- if user clicks "block" on initial permission request
image
- OR user is in incognito window
[Screencast from 2024-05-22 16-33-12.webm](https://github.com/BoltzExchange/boltz-web-app/assets/19181985/ba3ec599-7aae-4127-9c90-4047668a67d8)
Both resulting in permissions not granted, let's throw "Notification permissions not granted." error when user tries to enable here. Right now just nothing happens when trying to enable notifications in these states.

2. Notifications for all failed swaps, right now it seems there are no notifications for failed swap states:
Screenshot from 2024-05-22 16-37-05

**Review by kilrau — APPROVED** [review]
Previous review was addressed, tested in FF&Chromium and works as expected. Strings fixed.

**Review by michael1011 — CHANGES_REQUESTED** [review]

**Comment by kilrau** [comment]
> 2\. Notifications for all failed swaps, right now it seems there are no notifications for failed swap states:

Did you add notifications for failed swaps? I broke my regtest again and can't test rn.

**Comment by kilrau** [comment]
> 2\. Notifications for all failed swaps, right now it seems there are no notifications for failed swap states:

Moved into separate issue https://github.com/BoltzExchange/boltz-web-app/issues/601, from my side ok to merge this without

File: src/i18n/i18n.ts [hunk 1]
```diff
@@ -182,15 +183,19 @@ const dict = {
...
         logs: "Logs",
-        logs_tooltip: "Logs of the web app, useful for debugging.",
-        enable_audio_notifications: "Enable Audio Notifications",
+        logs_tooltip: "Logs of the web app, useful for debugging",
+        enable_audio_notifications: "Audio Notifications",
         enable_audio_notifications_tooltip:
             "Enable or disable audio notifications",
         on: "on",
         off: "off",
+        notification_header: "Boltz Swap",
```
Comment (michael1011) [inline src/i18n/i18n.ts:195]:
```suggestion
        notification_header: "Boltz",
```
I'd call this just Boltz. It's cleaner

**Review by michael1011 — APPROVED** [review]

File: src/i18n/i18n.ts (hunk 1 above)
Comment (kilrau) [reply src/i18n/i18n.ts]: Cleaned. For all languages.

**Pull request merged**